docs/guide/xlang_type_mapping.md

```diff
@@ -4,4 +4,76 @@ sidebar_position: 3
+| int32              | 4       | int/Integer     | int/pyfury.Int32     | Type.int32()    | int32_t                        | int32            | i32              |
+| var_int32          | 5       | int/Integer     | int/pyfury.VarInt32  | Type.varint32() | fury::varint32_t               | fury.varint32    | fury::varint32   |
+| int64              | 6       | long/Long       | int/pyfury.Int64     | Type.int64()    | int64_t                        | int64            | i64              |
+| var_int64          | 7       | long/Long       | int/pyfury.VarInt64  | Type.varint64() | fury::varint64_t               | fury.varint64    | fury::varint64   |
+| sli_int64          | 8       | long/Long       | int/pyfury.SliInt64  | Type.sliint64() | fury::sliint64_t               | fury.sliint64    | fury::sliint64   |
+| float16            | 9       | float/Float     | float/pyfury.Float16 | Type.float16()  | fury::float16_t                | fury.float16     | fury::f16        |
+| float32            | 10      | float/Float     | float/pyfury.Float32 | Type.float32()  | float                          | float32          | f32              |
+| float64            | 11      | double/Double   | float/pyfury.Float64 | Type.float64()  | double                         | float64          | f64              |
+| string             | 12      | String          | str                  | String          | string                         | string           | String/str       |
+| enum               | 13      | Enum subclasses | enum subclasses      | /               | enum                           | /                | enum             |
+| list               | 14      | List/Collection | list/tuple           | array           | vector                         | slice            | Vec              |
+| set                | 15      | Set             | set                  | /               | set                            | fury.Set         | Set              |
+| map                | 16      | Map             | dict                 | Map             | unordered_map                  | map              | HashMap          |
+| duration           | 17      | Duration        | timedelta            | Number          | duration                       | Duration         | Duration         |
+| timestamp          | 18      | Instant         | datetime             | Number          | std::chrono::nanoseconds       | Time             | DateTime         |
+| decimal            | 19      | BigDecimal      | Decimal              | bigint          | /                              | /                | /                |
+| binary             | 20      | byte[]          | bytes                | /               | `int8_t[n]/vector<T>`          | `[n]int8/[]T`    | `Vec<i8>`        |
+| array              | 20      | array           | np.ndarray           | /               | /                              | array/slice      | Vec              |
```

Comment:
There are many languages which have a dedicated type for binary instead of array. For example, python uses `bytes` for binary type. Binary us not used as an array in many cases. I suggest we keep it. There is also a binary type in Arrow.

Reply:
Thrift also has a binary type. And we don't provide uint8 type for now. The int8 array can be a different type

Reply:
We can serialize a bytes to int8 array and deserialize int8 array to bytes. But it would introduce some confusion. since it is so, we keep it. but should assign a new Typeid to binary to differentiate it from the array.

Reply:
> Thrift also has a binary type. And we don't provide uint8 type for now. The int8 array can be a different type

Yes I have noticed that we can not distinguish 255 from -1

Reply:
> We can serialize a bytes to int8 array and deserialize int8 array to bytes. But it would introduce some confusion. since it is so, we keep it. but should assign a new Typeid to binary to differentiate it from the array.
Yes  , I forgot to update the type Id. I will update it later